samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -404,6 +447,23 @@ class OffsetManager(
     }
   }
 
+  /**
+    * Load Startpoints for each SystemStreamPartition+TaskName
+    */
+  private def loadStartpoints: Unit = {
+    if (startpointManager != null) {
+      val taskNameToSSPs: Map[TaskName, Set[SystemStreamPartition]] = systemStreamPartitions
+
+      taskNameToSSPs.foreach {
+        case (taskName, systemStreamPartitionSet) => {
+          startpoints += taskName -> systemStreamPartitionSet
+            .map(ssp => (ssp, startpointManager.readStartpointForTask(ssp, taskName)))
```

Comment:
When we update the starting offset for a taskName using the defined startpoint instead of the checkpointed offset, it would be better to log it to aid debuggability.